cmd/kube-proxy/app/server.go

```diff
@@ -628,7 +632,24 @@ func newProxyServer(ctx context.Context, cfg *kubeproxyconfig.KubeProxyConfigura
 		return nil, err
 	}
 
-	rawNodeIPs := getNodeIPs(ctx, s.Client, s.Hostname)
+	// Make an informer that selects for our nodename.
+	currentNodeInformerFactory := informers.NewSharedInformerFactoryWithOptions(s.Client, s.Config.ConfigSyncPeriod.Duration,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector("metadata.name", s.Hostname).String()
+		}))
+	s.nodeInformer = currentNodeInformerFactory.Core().V1().Nodes()
+	nodeLister := s.nodeInformer.Lister()
+	nodeInformerHasSynced := s.nodeInformer.Informer().HasSynced
+
+	// This has to start after the calls to NewNodeConfig because that must
```

Comment:
this comment is removed in the next commit

Reply:
can you squash the commits together logically?